src/services/goals.js

```diff
@@ -358,19 +358,33 @@ export function reduceObjectives(newObjectives, currentObjectives = []) {
 
     const id = objective.getDataValue('id') ? objective.getDataValue('id') : objective.getDataValue('value');
 
+    const arOrder = objective.activityReportObjectives
+      && objective.activityReportObjectives[0]
+      && objective.activityReportObjectives[0].arOrder
+      ? objective.activityReportObjectives[0].arOrder : null;
+
     return [...objectives, {
       ...objective.dataValues,
       value: id,
       ids: [id],
       // Make sure we pass back a list of recipient ids for subsequent saves.
       recipientIds: [objective.getDataValue('otherEntityId')],
       isNew: false,
+      arOrder,
     }];
   }, currentObjectives);
+
+  const sortedObjectives = objectivesToSort.sort((o1, o2) => {
```

Comment:
```sort``` sorts the array in place, so it might make sense to just sort and return without saving to a variable